specs/order_spec.rb

```diff
@@ -76,35 +84,119 @@
       result.must_equal true
     end
   end
+
+  #Basically the opposite of #add_product
+ describe "#remove_product" do
+  it "Decreases the number of products" do
+   products = { "banana" => 1.99, "cracker" => 3.00 }
+   before_count = products.count
+   order = Grocery::Order.new(1337, products)
+
+   # based only on the first argument or name of product
+   order.remove_product("banana")
+   # remove instead of add
+   expected_count = before_count - 1
+   order.products.count.must_equal expected_count
+  end
+
+  it "Is removed to the collection of products" do
+   products = { "banana" => 1.99, "cracker" => 3.00 }
+   order = Grocery::Order.new(1337, products)
+
+   order.remove_product("sandwich")
+   order.products.include?("sandwich").wont_equal true
+  end
+
+  it "Returns true if the product is removed" do
+   products = { "banana" => 1.99, "cracker" => 3.00 }
+
+   order = Grocery::Order.new(1337, products)
+   before_total = order.total
+
+   result = order.remove_product("banana")
+   after_total = order.total
+
+   # Returns true if the before total != after total when item is removed
+   result.must_equal true
+   before_total.wont_equal after_total
+  end
+
+  it "Returns false if the product is new" do
+   products = { "banana" => 1.99, "cracker" => 3.00 }
+   order = Grocery::Order.new(1337, products)
+
+   result = order.remove_product("salad")
+   result.must_equal false
+  end
+
+ end
 end
 
 # TODO: change 'xdescribe' to 'describe' to run these tests
-xdescribe "Order Wave 2" do
+describe "Order Wave 2" do
   describe "Order.all" do
+
     it "Returns an array of all orders" do
-      # TODO: Your test code here!
+      # wait don't I need to put an "arrange" statement here??
+
+      all_orders = Grocery::Order.all
+      # all_orders.must_be_kind_of Array
+      all_orders.class.must_equal Array
+
+      all_orders.each do |order|
+        order.must_be_instance_of Grocery::Order
+      end
+
+      all_orders.count.must_equal 100
     end
 
     it "Returns accurate information about the first order" do
-      # TODO: Your test code here!
+
+      all_orders = Grocery::Order.all
+      # binding.pry
+
+
+      all_orders[0].id.must_equal 1
+
+      all_orders[0].products.must_be_kind_of Hash
+
+
     end
 
     it "Returns accurate information about the last order" do
-      # TODO: Your test code here!
+      all_orders = Grocery::Order.all
+
+      all_orders[-1].id.must_equal 100
+      all_orders[-1].products.must_be_kind_of Hash
+
+
     end
   end
 
   describe "Order.find" do
+
+  before do
+    @order_class = Grocery::Order
+  end
+
+
     it "Can find the first order from the CSV" do
-      # TODO: Your test code here!
+
+      @order_class.find(1).must_be_instance_of Grocery::Order
```

Comment:
You should also check to ensure the values of the found Order have the right values, check the `id`, `products` etc.